scripts/rh-manifest.sh

```diff
@@ -7,8 +7,15 @@ set -u
 
 WORKING_FILE="$(mktemp /tmp/rh-manifest.XXXXXXXX)"
 
-find . -name package.json -execdir yarn list --silent --json \; | \
-	jq -r '.. | objects | .name? | select(. != null)' \
-	   2>/dev/null >>"${WORKING_FILE}"
+atexit() {
+	echo "Cleaning up..."
+	rm "$WORKING_FILE"
+}
 
-sort "${WORKING_FILE}" | uniq > rh-manifest.txt
+trap atexit EXIT
+
+find . -name package-lock.json -execdir npm list --prod --json --all --package-lock-only \; |
```

Comment:
Could you explain why using lock file and npm (instead of yarn) solved the issue?

Reply:
because yarn complains about missing lock file and not able to produce a list of dependencies.
```bash
➜  thanos git:(master) ./scripts/rh-manifest.sh
{"type":"error","data":"No lockfile in this directory. Run `yarn install` to generate one."}
```

Reply:
Ok, thanks, understand better now.